Update the CSS and some controls for the CORS page:

michellephung opened the pull request:
- the 'update' input field for editing a domain name, now automatically
resizes and no longer overlaps the pencil/trashcan buttons
- users can hit 'enter' to validate and submit the 'Update' domain  and
'Add' domain fields
- page does not return an empty origin, if "specific origins" is set, but
 there are no domains on that list

closes COUCHDB-2569

garrensmith commented on app/addons/cors/components.react.jsx:
```diff
@@ -232,7 +251,7 @@ define([
     enableCorsChange: function (event) {
       if (this.state.corsEnabled && !_.isEmpty(this.state.origins) && !this.state.isAllOrigins) {
-        var result = window.confirm('Are you sure? Disabling CORS will overwrite your specific origin domains.');
+        var result = window.confirm('Are you sure? Disabling CORS will disable browser access to your databases from other applications.');
```
I prefer the original message here.

garrensmith commented:
Looking good @michellephung. Can you write a test for the "page does not return an empty origin, if "specific origins" is set, but there are no domains on that list" and can you fix the failing test.

michellephung commented:
test is working :D

michellephung commented:
@robertkowalski @garrensmith 

I updated the onKeyDown to onKeyUp, per Robert's suggestion.

Still not sure if I did the test right. I moved the delete-empty-string-for-origin into cors/resources.js before we set the stores. So I moved the test into resourceSpec.js.

benkeen commented on app/addons/cors/components.react.jsx:
```diff
@@ -69,11 +69,17 @@ define([
     },
 
+    onKeyUp: function (e) {
+      if (e.keyCode == 13 ) {   //enter key
```
extra space after 13

benkeen commented on app/addons/cors/components.react.jsx:
```diff
@@ -152,6 +158,12 @@ define([
     },
 
+    onKeyUp: function (e) {
+      if (e.keyCode == 13 ) {   //enter key
```
same here

robertkowalski commented:
hi michelle, the test is really cool but the way `parse` is used is not quite right.

robertkowalski commented:
after the parse the (modified) contents from the server (empty string changed to an empty array: `[]`) the object is passed internally to something like a set which populates the properties on the model.

see also http://backbonejs.org/#Model-parse:

```
The function is passed the raw response object, and should return 
the attributes hash to be set on the model. 
```

robertkowalski commented:
This way you always have the right properties already set on your model, no matter which way to get the value you use.

robertkowalski commented:
merged as b29ad53cfc3df39dd91412ec01c54d75be6ceab6 - thank you michelle!
